packages/roosterjs-content-model-plugins/lib/tableEdit/editors/features/TableMover.ts

```diff
@@ -129,3 +157,214 @@ function isTableTopVisible(editor: IEditor, rect: Rect | null, contentDiv?: Node
+        if (pos && domHelper.isNodeInEditor(pos.offsetNode)) {
+            return { node: pos.offsetNode, offset: pos.offset };
+        }
+    }
+
+    if (doc.elementFromPoint) {
+        // Fallback
+        const element = doc.elementFromPoint(x, y);
+        if (element && domHelper.isNodeInEditor(element)) {
+            return { node: element, offset: 0 };
+        }
+    }
+
+    return null;
+}
+
+function onDragging(context: TableMoverContext, event: MouseEvent, initValue: TableMoverInitValue) {
+    const { tableRect } = initValue;
+    const { editor } = context;
+
+    // Move table outline rectangle
+    tableRect.style.top = `${event.clientY + TABLE_MOVER_LENGTH}px`;
+    tableRect.style.left = `${event.clientX + TABLE_MOVER_LENGTH}px`;
+
+    event.ctrlKey || event.metaKey
+        ? setTableMoverCursor(editor, true, 'copy')
+        : setTableMoverCursor(editor, true, 'move');
+
+    const pos = getNodePositionFromEvent(editor, event.clientX, event.clientY);
+    if (pos) {
+        const range = editor.getDocument().createRange();
+        range.setStart(pos.node, pos.offset);
+        range.collapse(true);
+
+        editor.setDOMSelection({ type: 'range', range, isReverted: false });
+        return true;
+    }
+    return false;
+}
+
+function onDragEnd(
+    context: TableMoverContext,
+    event: MouseEvent,
+    initValue: TableMoverInitValue | undefined
+) {
+    const { editor, table, onFinishDragging: selectWholeTable } = context;
+    const element = event.target;
+    // Remove table outline rectangle
+    initValue?.tableRect.remove();
+
+    setTableMoverCursor(editor, false);
+
+    if (element == context.div) {
+        // Table mover was only clicked, select whole table
+        selectWholeTable(table);
+    } else {
+        // Check if table was dragged on itself or Check if element is not in editor
+        if (
+            table.contains(element as Node) ||
+            !editor.getDOMHelper().isNodeInEditor(element as Node)
+        ) {
+            editor.setDOMSelection(initValue?.initialSelection ?? null);
+            return cancelMove(context);
+        }
+
+        const finalRange = editor.getDOMSelection();
+        if (initValue && finalRange?.type == 'range') {
+            // Move table to new position
+            finalRange.range.collapse(true);
+            let insertionSuccess: boolean = false;
+
+            // Insert table into content model
+            editor.formatContentModel(
+                (model, context) => {
+                    const SPArray = getSelectedSegmentsAndParagraphs(model, false);
+                    if (
+                        SPArray.length == 1 &&
+                        SPArray[0][0].segmentType == 'SelectionMarker' &&
+                        SPArray[0][1] != null
+                    ) {
+                        const paragraph = SPArray[0][1];
+                        paragraph.segments.forEach(segment => {
```

Comment:
can we use mergeModel here?

Reply:
Yes, I think I was trying to check for the exact segment to do the insertion but it was unnecessary.